deploy/charts/README.md

```diff
@@ -0,0 +1,193 @@
+
+Helm Deployment
+===============
+
+Prerequisites
+-------------
+
+### Kubernetes
+
+You will need Kubernetes installed. (Docker for
+Desktop)[https://www.docker.com/products/docker-desktop] includes a
+workable one-node Kubernetes installation - just need to activate it
+in the settings. Alternatively, you can install
+(Minikube)[https://kubernetes.io/docs/tasks/tools/install-minikube/].
+
+Note that on both Windows and Mac, this will run containers within a
+VM that has limited resources. You might want to increase this to at
+least 3 GB. For Docker this can be found in the settings, for Minikube
+you need to specify it on the command line:
+
+    $ minikube --mem 4096 ...
+
+### Helm
+
+Furthermore you will need to install the Helm utility. It is available
+from most typical package managers, see (Using
+Helm)[https://helm.sh/docs/using_helm/]. Note that for the moment we
+are using Helm version 2, as at the time of writing version 3 is in
+early alpha.
+
+Once you have it available, you will typically need to initialise it
+(this will create the "Tiller" controller):
+
+    $ helm --init
+
+Deploying SDP
+-------------
+
+### Creating etcd-operator
+
+We are not creating an etcd cluster ourselves, but instead leave it to
+"operator" that needs to be installed first. Simply execute:
+
+    $ helm install stable/etcd-operator -n etcd
+
+If you now execute:
+
+    $ kubectl get pod --watch
+
+You should eventually see an pod called
+`etcdop-etcd-operator-etcd-operator-[...]` in "Running" state (yes,
+Helm is exceedingly redundant with its names). If not wait a bit, if
+you try to go to the next step before this has completed there's a
+chance it will fail.
+
+### Deploy the prototype
+
+At this point you should be able to deploy
+
+    $ cd [sdp-prototype]/deploy/charts
+    $ helm install sdp-prototype -n sdp-prototype
+
+You can again watch the fireworks using `kubectl`:
+
+    $ kubectl get pod --watch
+
+Pods asocciated with Tango might go down a couple times before they
+start correctly, this seems to be normal. You can check the logs of
+pods (copy the full name from `kubectl` output) to verify that they
+are doing okay:
+
+    $ kubectl logs sdp-prototype-workflow-testdeploy-[...]
+    INFO:main:Waiting for processing block...
+    $ kubectl logs sdp-prototype-helm-[...]
+    ...
+    INFO:main:Found 0 existing deployments.
+    $ kubectl logs sdp-protoype-sdp-master-[...]
+    ...
+    Ready to accept request
+
+Just to name a few. If it's looking like this, there's a good chance
+everything deployed correctly.
+
+Testing it out
+--------------
+
+### Connecting to configuration database
+
+The deployment scripts should have exposed the SDP configuration
+database (i.e. etcd) via a NodePort service. For Minikube, you can
```

Comment:
How do you do this if you're not using Minikube? (By default on Mac, you're running a full k8s environemtn.